test/viewmodels/checkAnswers/sections/firstTransporter/FirstTransporterHasVatSummarySpec.scala

```diff
@@ -0,0 +1,103 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package viewmodels.checkAnswers.sections.firstTransporter
+
+import base.SpecBase
+import fixtures.messages.sections.firstTransporter.FirstTransporterVatMessages
+import models.CheckMode
+import models.sections.firstTransporter.FirstTransporterVatModel
+import org.scalatest.matchers.must.Matchers
+import pages.sections.firstTransporter.FirstTransporterVatPage
+import play.api.i18n.{Messages, MessagesApi}
+import play.api.test.FakeRequest
+import uk.gov.hmrc.govukfrontend.views.Aliases.Value
+import uk.gov.hmrc.govukfrontend.views.viewmodels.content.Text
+import viewmodels.govuk.summarylist._
+import viewmodels.implicits._
+
+
+class FirstTransporterHasVatSummarySpec extends SpecBase with Matchers {
+  "FirstTransporterHasVatSummary" - {
+    Seq(FirstTransporterVatMessages.English).foreach { messagesForLanguage =>
+
+      s"when being rendered in lang code of '${messagesForLanguage.lang.code}'" - {
+
+        implicit val msgs: Messages = app.injector.instanceOf[MessagesApi].preferred(Seq(messagesForLanguage.lang))
+
+        "when the show action link boolean is true" - {
+
+          "when there is no answer" in {
+            implicit lazy val request = dataRequest(FakeRequest(), emptyUserAnswers)
+
+            FirstTransporterHasVatSummary.row(showActionLinks = true) mustBe SummaryListRowViewModel(
+              key = messagesForLanguage.hasVatCyaLabel,
+              value = Value(Text("Not provided")),
+              actions = Seq(
+                ActionItemViewModel(
+                  content = messagesForLanguage.change,
+                  href = controllers.sections.firstTransporter.routes.FirstTransporterVatController.onPageLoad(testErn, testDraftId, CheckMode).url,
+                  id = "changeFirstTransporterVat"
+                ).withVisuallyHiddenText(messagesForLanguage.cyaChangeHidden)
+              )
+            )
+          }
+
+          "when there is an answer" in {
+            implicit lazy val request = dataRequest(FakeRequest(), emptyUserAnswers.set(FirstTransporterVatPage, FirstTransporterVatModel(true, Some(testVatNumber))))
+
+            FirstTransporterHasVatSummary.row(showActionLinks = true) mustBe SummaryListRowViewModel(
+              key = messagesForLanguage.hasVatCyaLabel,
+              value = Value(Text("Yes")),
+              actions = Seq(
+                ActionItemViewModel(
+                  content = messagesForLanguage.change,
+                  href = controllers.sections.firstTransporter.routes.FirstTransporterVatController.onPageLoad(testErn, testDraftId, CheckMode).url,
+                  id = "changeFirstTransporterVat"
+                ).withVisuallyHiddenText(messagesForLanguage.cyaChangeHidden)
+              )
+            )
+          }
+        }
+
+        "when the show action link boolean is false" - {
```

Comment:
I'm not sure why we have this tbf, currently it's always is true and can't be changed. Nothing to do with your changes but could be refactored 😄